frame/dex/src/lib.rs

```diff
@@ -307,7 +293,7 @@ pub mod pallet {
 				}
 
 				let pallet_account = Self::account_id();
-				T::Assets::transfer(asset1, &sender, &pallet_account, amount1, false)?;
+				T::Currency::transfer(&sender, &pallet_account, amount1, false)?;
```

Comment:
I'm wondering if that would actually work, as the last param in the transfer method for native currency is an enum

Reply:
It's this trait that it's using now to do the transfer:
https://github.com/paritytech/substrate/blob/f0b6e79a8d21856743acc5370b5d5c3271048cb9/frame/support/src/traits/tokens/fungible.rs#L123